src/CommonLib/Processors/LDAPPropertyProcessor.cs

```diff
@@ -589,6 +489,167 @@ public async Task<ComputerProperties> ReadComputerProperties(ISearchResultEntry
 
             return props;
         }
+        
+        public static Dictionary<string, object> GetProperties(string ldapProperty, ISearchResultEntry entry)
+        {
+            var props = new Dictionary<string, object>();
+            switch (ldapProperty)
+            {
+                case LDAPProperties.Description:
+                    props.Add(PropertyMap.GetPropertyName(ldapProperty), entry.GetProperty(LDAPProperties.Description));
+                    break;
+                case LDAPProperties.WhenCreated:
+                    props.Add(PropertyMap.GetPropertyName(ldapProperty), Helpers.ConvertTimestampToUnixEpoch(entry.GetProperty(LDAPProperties.WhenCreated)));
+                    break;
+                case LDAPProperties.DomainFunctionalLevel:
+                    if (!int.TryParse(entry.GetProperty(LDAPProperties.DomainFunctionalLevel), out var level))
+                        level = -1;
+                    props.Add(PropertyMap.GetPropertyName(ldapProperty), FunctionalLevelToString(level));
+                    break;
+                case LDAPProperties.GPCFileSYSPath:
+                    props.Add(PropertyMap.GetPropertyName(ldapProperty), entry.GetProperty(LDAPProperties.GPCFileSYSPath)?.ToUpper());
+                    break;
+                case LDAPProperties.LastLogon:
+                    props.Add(PropertyMap.GetPropertyName(ldapProperty), Helpers.ConvertFileTimeToUnixEpoch(entry.GetProperty(LDAPProperties.LastLogon)));
+                    break;
+                case LDAPProperties.LastLogonTimestamp:
+                    props.Add(PropertyMap.GetPropertyName(ldapProperty), Helpers.ConvertFileTimeToUnixEpoch(entry.GetProperty(LDAPProperties.LastLogonTimestamp)));
+                    break;
+                case LDAPProperties.PasswordLastSet:
+                    props.Add(PropertyMap.GetPropertyName(ldapProperty), Helpers.ConvertFileTimeToUnixEpoch(entry.GetProperty(LDAPProperties.PasswordLastSet)));
+                    break;
+                case LDAPProperties.ServicePrincipalNames:
+                    var spn = entry.GetArrayProperty(LDAPProperties.ServicePrincipalNames);
+                    props.Add(PropertyMap.GetPropertyName(ldapProperty), spn);
+                    props.Add("hasspn", spn.Length > 0);
```

Comment:
I don't like creating a precedent here for wild-west property names (that aren't vetted through `PropertyMap`) from the get out of a refactor, but I have no better solutions for this many-to-one property map at the moment.
Because we're still vetting through our single-truth mapping method I'm inclined to let it slide, but I might be the only one.